Client/Frontend/UserContent/UserScripts/FarblingProtection.js

```diff
@@ -0,0 +1,304 @@
+      newPlugin[newMimeType.type] = newMimeType
+
+      Reflect.defineProperty(newMimeType, 'enabledPlugin', {
+        get: function () {
+          return newPlugin
+        }
+      })
+    }
+
+    // Fix .item() function to return the correct item
+    newPlugin.item = function (index) {
+      return newPlugin[index]
+    }
+
+    return newPlugin
+  }
+
+  // We need the original length so we can reference it (as we will change it)
+  const plugins = window.navigator.plugins
+  const originalPluginsLength = plugins.length
+  const originalItemFunction = plugins.item
+
+  // Adds a fake plugin for the given index on fakePluginData
+  const addPluginAtIndex = (newPlugin, index) => {
+    const pluginPosition = originalPluginsLength + index
+    window.navigator.plugins[pluginPosition] = newPlugin
+    window.navigator.plugins[newPlugin.name] = newPlugin
+  }
+
+  for (const [index, pluginData] of fakePluginData.entries()) {
+    const newPlugin = makeFakePlugin(pluginData)
+    addPluginAtIndex(newPlugin, index)
+  }
+
+  // Adjust the length of the original plugin array
+  Reflect.defineProperty(window.navigator.plugins, 'length', {
+    get: function () {
+      return originalPluginsLength + fakePluginData.length
+    }
+  })
+
+  // Fix .item() function to return the correct item
+  window.PluginArray.prototype.item = function (index) {
+    if (index < originalPluginsLength) {
+      return Reflect.apply(originalItemFunction, plugins, arguments)
+    } else {
+      return plugins[index]
+    }
+  }
+
+  // 4. Farble speech synthesizer
+  const makeFakeVoiceFromVoice = (voice) => {
+    const voicePrototype = Object.getPrototypeOf(voice)
+
+    const newVoice = Object.create(voicePrototype, {
+      name: {
+        get: function () {
+          return fakeVoiceName
+        }
+      },
+      voiceURI: {
+        get: function () {
+          return voice.voiceURI
+        }
+      },
+      lang: {
+        get: function () {
+          return voice.lang
+        }
+      },
+      localService: {
+        get: function () {
+          return voice.localService
+        }
+      },
+      default: {
+        get: function () {
+          return false
+        }
+      }
+    })
+
+    return newVoice
+  }
+
+  let originalVoice
+  let fakeVoice
+  let passedFakeVoice
+
+  // We need to override the voice property to allow our fake voice to work
+  const descriptor = Reflect.getOwnPropertyDescriptor(SpeechSynthesisUtterance.prototype, 'voice')
+  Reflect.defineProperty(SpeechSynthesisUtterance.prototype, 'voice', {
+    get: function () {
+      if (passedFakeVoice === undefined) {
+        // We didn't set a fake voice
+        return Reflect.apply(descriptor.get, this, arguments)
+      } else {
+        // We set a fake voice, return that instead
+        return passedFakeVoice
+      }
+    },
+    set: function (passedVoice) {
+      if (passedVoice === fakeVoice && originalVoice !== undefined) {
```

Comment:
Will `originalVoice` be guaranteed `undefined` or is it possible to be `null` too?

Reply:
I never nullify `originalVoice`. I also never nullify `fakeVoice`